.github/CHANGELOG.md

```diff
@@ -6,14 +6,22 @@
 - Introduced the `BaseEngine` abstract base class and the `LocalEngine` child class. `Engine` is kept as an alias for `LocalEngine`.
 - The Engine API has been changed slightly:
     - `LocalEngine.run()` returns a `Result` object that contains both a state object and measurement samples.
-    - The way kwargs were used has been simplified by introducing the new kwargs-like arguments `backend_options` and `state_options`. `LocalEngine.run()` passes the actual kwargs only to `Operation.apply()`.
-- The Gaussian backend now officially supports Fock-basis measurements (`MeasureFock`/`Measure`/`measure_fock`), but does not update the quantum state.
-- New `shots` keyword argument added to `Engine.run()`, enabling multi-shot sampling. Supported only in the Gaussian backend, and only for Fock measurements.
-- Added the ability to compile quantum programs to match a desired circuit target.
-- Included a number of compilation targets, including Gaussian Boson Sampling circuits.
-- Added a frontend validation database, the `devicespecs` submodule, for validating that quantum programs can be executed on certain backends and providing compilation methods.
-- Added the `sf.io` module, which is used to save/load standalone Blackbird scripts from/into Strawberry Fields. Note that the Blackbird DSL has been spun off as an independent package and is now a dependency of Strawberry Fields.
-- Added a new decomposition `mach_zehnder` to the decompositions module. 
+    - The way kwargs were used has been simplified by introducing the new kwargs-like arguments:
+        - `backend_options` are provided upon initialization of the `LocalEngine`, and passed on to the corresponding backend when it is created.
+        - `compile_options` can be provided when calling `LocalEngine.run()`. These are passed to the `compile()` method of the program before execution.
+        - `run_options` can be provided when calling `LocalEngine.run()`. These are passed to the backend API calls via :meth:`Operation.apply`
+        - `state_options` can be provided when calling `LocalEngine.run()`. These are used to determine the characteristic of the `Results` object returned after the program is finished executing.
```

Comment:
Feeling like `state_options` should be combined with `run_options`. The only things that `run_options` accepts are various TF parameters, which are needed in order to determine what types to return for measurement results and final states.

Reply:
I support all of the above :+1: So this will result in two dicts, `compile_options` and `run_options`? I'm in favour of calling it `run_options`, simply because some of the keys are used during the engine run.

Reply:
Done in 60221ad